src/index.js

```diff
@@ -0,0 +1,155 @@
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = task.completed;
+    checkbox.addEventListener('change', () => {
+      task.completed = checkbox.checked;
+      renderTasks();
+    });
+
+    const descriptionSpan = document.createElement('span');
+    descriptionSpan.classList.add('description'); // Add class for the description span
+    descriptionSpan.textContent = task.description;
+
+    // Create the edit icon
+    const editIcon = document.createElement('span');
+    editIcon.classList.add('edit-icon');
+    editIcon.textContent = '✏️';
+    // eslint-disable-next-line no-use-before-define
+    editIcon.addEventListener('click', () => editTask(task.index));
+
+    const inputElement = document.createElement('input');
+    inputElement.type = 'text';
+    inputElement.classList.add('edit-input');
+    inputElement.style.display = 'none';
+
+    const removeButton = document.createElement('button');
+    removeButton.textContent = 'Remove';
+    // eslint-disable-next-line no-use-before-define
+    removeButton.addEventListener('click', () => removeTask(task.index));
+
+    listItem.appendChild(checkbox);
+    listItem.appendChild(descriptionSpan);
+    listItem.appendChild(editIcon);
+    listItem.appendChild(inputElement);
+    listItem.appendChild(removeButton);
+    taskList.appendChild(listItem);
+  });
+
+  // Store tasks in local storage
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+// Function to add a new task
+function addTask(description) {
+  const index = tasks.length > 0 ? tasks[tasks.length - 1].index + 1 : 1;
+  tasks.push({ description, completed: false, index });
+  renderTasks();
+}
+
+// Function for updating the indexes.
+function updateTaskIndexes() {
+  tasks.forEach((task, index) => {
+    task.index = index + 1;
+  });
+}
+
+// Function to remove a task
+function removeTask(index) {
+  const taskIndex = tasks.findIndex((task) => task.index === index);
+  if (taskIndex !== -1) {
+    tasks.splice(taskIndex, 1);
+    updateTaskIndexes();
+    renderTasks();
+  }
+}
+
+// Function to edit a task
+function editTask(index) {
+  const taskIndex = tasks.findIndex((task) => task.index === index);
+  if (taskIndex !== -1) {
+    const listItem = document.getElementById(`taskList-${index}`);
+    const descriptionSpan = listItem.querySelector('.description');
+    const editIcon = listItem.querySelector('.edit-icon');
+    const inputElement = listItem.querySelector('.edit-input');
+
+    // Show the input element and set its value to the task description
+    inputElement.style.display = 'inline-block';
+    inputElement.value = tasks[taskIndex].description;
+    inputElement.focus();
+
+    // Hide the description span and edit icon during editing
+    descriptionSpan.style.display = 'none';
+    editIcon.style.display = 'none';
+
+    // Save changes when the input loses focus
+    // eslint-disable-next-line no-use-before-define
```

Comment:
- [ ]  Kindly make sure you do not disable linters rules for they help you to keep your code clean and conform to the universal standards